src/rf/apps/workers/process.py

```diff
@@ -0,0 +1,221 @@
+
+    def start_polling(self):
+        while True and self.q:
+            messages = self.q.get_messages(message_attributes=['All'])
+            if len(messages) == 0:
+                # If nothing returned from the queue, wait a little bit more.
+                self.wait = self.wait + 1 if self.wait < MAX_WAIT else MAX_WAIT
+                time.sleep(self.wait)
+            elif len(messages) == 1:
+                self.wait = MIN_WAIT
+                delete_message = True  # Default to true.
+                # If we get an unexpected or improperly formatted message, we
+                # want to remove it from the queue.
+                message = messages[0]
+                body = json.loads(message.get_body())
+                if ('Records' in body and body['Records'][0]):
+                    record = body['Records'][0]
+                    type = record['eventName']
+                    if type == JOB_REPROJECT:
+                        delete_message = self.reproject(record['data'])
+                    elif type == JOB_HANDOFF:
+                        delete_message = self.emr_hand_off(record['data'])
+                    elif type == JOB_TIMEOUT:
+                        delete_message = self.check_timeout(record['data'])
+                    elif type in JOB_VALIDATE:
+                        if record['eventSource'] != 'aws:s3':
+                            return
+
+                        delete_message = self.validate_image(record)
+
+                if delete_message:
+                    self.q.delete_message(message)
+
+    def validate_image(self, data):
+        """
+        Use Gdal to verify an image is properly formatted and can be further
+        processed.
+        data -- attribute data from SQS.
+        """
+        # Pass image to Gdal to verify.
+        # If it succeeds continue else return False.
+
+        key = data['s3']['object']['key']
+        dot = key.find('.') if key.find('.') >= 0 else len(key)
+        first_dash = key.find('-')
+        payload_uuid = key[first_dash:dot]
+
+        url = self.make_s3_url(data['s3']['bucket']['name'], key)
+
+        # TODO - Right now we don't save data to the DB with upload. So hard
+        # code something for testing.
+        try:
+            image = LayerImage.objects.get(s3_uuid=payload_uuid)
+            image.status = STATUS_VALIDATED
+            image.save()
+        except:
+            payload_uuid = '1aa064aa-1086-4ff1-a90b-09d3420e0343'
+
+        data = {'url': url, 's3_uuid': payload_uuid}
+        payload = self.make_payload(JOB_REPROJECT, data)
+        self.post_to_queue(payload)
+        return True
+
+    def reproject(self, data):
+        """
+        Reproject incoming image to Web Mercator.
+        data -- attribute data from SQS.
+        """
+        # Pass image to Gdal to reproject
+        # If it succeeds continue else return False.
+
+        s3_uuid = data['s3_uuid']
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
```

Comment:
I think `s3_uuid` and `source_uri` should be combined. It seems like `source_uri` should already contain a uuid but it looks like that hasn't been implemented yet.